src/core/server/mocks.ts

```diff
@@ -112,12 +113,19 @@ function createCoreSetupMock() {
   const uiSettingsMock = {
     register: uiSettingsServiceMock.createSetupContract().register,
   };
+
+  const savedObjectsService = savedObjectsServiceMock.createSetupContract();
+  const savedObjectMock: jest.Mocked<CoreSetup['savedObjects']> = {
+    addClientWrapper: savedObjectsService.addClientWrapper,
+    setClientFactoryProvider: savedObjectsService.setClientFactoryProvider,
+  };
```

Comment:
Many services follow a pattern where we define a `serviceMock.createInternalSetupContract` and a `serviceMock.createSetupContract` to make it a bit more clear what the differences are.

Reply:
I agree, this is more explicit. We should probably do this on every service for consistency. Doing the change for the soService mock.